tests/base.py

```diff
@@ -114,6 +114,22 @@ def expected_metadata(self):
                 self.REPLICATION_METHOD: self.INCREMENTAL,
                 self.REPLICATION_KEYS: {"modified_at"},
                 self.EXPECTED_PAGE_SIZE: 50
+            },
+            "happiness_ratings": {
+                self.PRIMARY_KEYS: {"rating_customer_id", "conversation_id", "rating_created_at"},
+                self.REPLICATION_METHOD: self.FULL_TABLE,
```

Comment:
Why exactly is this full table, especially if the request (built in abstract.py) is pulling the start parameter from state? Is there not a reliable means of incrementing the requests in that manner?

Reply:
Customer has the ability to edit the rating but there is no field like `updatedAt` or `modifiedAt` in the response [schema](https://developer.helpscout.com/mailbox-api/endpoints/reports/happiness/reports-happiness-ratings/#response-fields) which says when rating was updated by the customer and value of ratingCreatedAt field doesn't change if the customer edits the rating(I have verified it myself with test account).
If we implement `INCREMENTAL` method then we end up missing all the edited records and surprisingly, the endpoint doesn't work without passing the `start` and `end` QueryParams.So, start param always takes the `start_date` value set by our client as default since we don't store the bookmark for this stream and end param takes current_time.

Reply:
Gotcha. We should keep an eye on this in the future in case they add some kind of limit for how long users can modify the ratings, in which case we could swap to doing a lookback window for that time period instead.